privval/signer_server.go

```diff
@@ -72,16 +72,12 @@ func (ss *SignerServer) servicePendingRequest() {
 		return
 	}
 
-	var res privvalproto.Message
-	{
-		// limit the scope of the lock
-		ss.handlerMtx.Lock()
-		defer ss.handlerMtx.Unlock()
-		res, err = ss.validationRequestHandler(ss.privVal, req, ss.chainID)
-		if err != nil {
-			// only log the error; we'll reply with an error in res
-			ss.Logger.Error("SignerServer: handleMessage", "err", err)
-		}
+	ss.handlerMtx.Lock()
+	res, err := ss.validationRequestHandler(ss.privVal, req, ss.chainID)
+	ss.handlerMtx.Unlock()
```

Comment:
`validationRequestHandler` is a functor whose default value is `DefaultValidationRequestHandler`

I can see that `DefaultValidationRequestHandler` can panic. It is not go-idiomatic to leave a mutex locked in case of panic (the panic might be "recovered" up in the call stack).

This should rather read:
```go
func () {
	ss.handlerMtx.Lock()
	defer ss.handlerMtx.Unlock()
	res, err := ss.validationRequestHandler(ss.privVal, req, ss.chainID)
}()
```
Check [this](https://go.dev/play/p/OrlUdJ5ZZ05) for an example.

Reply:
Thanks for the knowledge, sir. I have opened a new PR #3135 for this😀

Reply:
Thanks a lot! Taking a look